ios/Video/CustomPlayerController/RCTPlaybackController.swift

```diff
@@ -616,12 +635,43 @@ class RCTPlaybackController: UIView, AVRoutePickerViewDelegate {
         self.curTimeLabel.text = secondsToTimeLabel(seconds)
     }
     
-    func setUi_durationTime(seconds: Float){
-        if(_isLive){
-            self.durTimeLabel.text = "Live"
-        }else{
-            self.durTimeLabel.text = secondsToTimeLabel(seconds)
+    func setTimeLabelSize(isHours: Bool){
+        let width = isHours ? TIME_LABEL_SIZE_HOURS: TIME_LABEL_SIZE_MINUTES
+        let curTimeWidth = _isLive ? width + TIME_LABEL_SIZE_LIVE_OFFSET : width
+        
+        // Create width constraint for current time label if it doesn't exist
+        if curTimeWidthConstraint == nil {
+            curTimeWidthConstraint = NSLayoutConstraint(item: curTimeLabel, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 0)
+            curTimeLabel.addConstraint(curTimeWidthConstraint!)
+        }
+        
+        // Create width constraint for duration label if it doesn't exist and it's not a live stream
+        if durTimeWidthConstraint == nil && !_isLive {
```

Comment:
This function feels fairly complex. You could simplify it a bit by always creating the two constraints and just update them when necessary (line 661 - 665). For live streams you could keep certain constrains `nil` and the optional chaining will then simply ignore it.

Reply:
I'm now just disabling the constraint instead